src/vs/workbench/contrib/localization/electron-sandbox/localeService.ts

```diff
@@ -26,11 +30,33 @@ export class NativeLocaleService implements ILocaleService {
 		@ILanguagePackService private readonly languagePackService: ILanguagePackService,
 		@IPaneCompositePartService private readonly paneCompositePartService: IPaneCompositePartService,
 		@IExtensionManagementService private readonly extensionManagementService: IExtensionManagementService,
-		@IProgressService private readonly progressService: IProgressService
+		@IProgressService private readonly progressService: IProgressService,
+		@ICommandService private readonly commandService: ICommandService,
+		@ITextFileService private readonly textFileService: ITextFileService
 	) { }
 
-	private async writeLocaleValue(locale: string | undefined): Promise<void> {
+	private async writeLocaleValue(locale: string | undefined): Promise<boolean> {
+		try {
+
+			const content = await this.textFileService.read(this.environmentService.argvResource, { encoding: 'utf8' });
+
+			// This is the same logic that we do where argv.json is parsed so mirror that:
+			// https://github.com/microsoft/vscode/blob/32d40cf44e893e87ac33ac4f08de1e5f7fe077fc/src/main.js#L238-L246
+			JSON.parse(stripComments(content.value));
+		} catch (error) {
+			this.notificationService.notify({
+				severity: Severity.Error,
+				message: localize('argvInvalid', 'Your argv.json file is not valid. Please open it and fix any parse errors so that your display language can be set properly.'),
```

Comment:
Maybe we can make this more user friendly: the fact that we store the locale in a file `argv.json` is not so important imho. In the end I would just indicate that the settings file where we store the locale is malformed and needs manual fixing.

I think settings editor has a similar error notification, maybe we can borrow the wording.